pkg/osutil/interrupt_unix.go

```diff
@@ -69,6 +69,11 @@ func HandleInterrupts() {
 			os.Exit(0)
 		}
 		syscall.Kill(pid, sig.(syscall.Signal))
+
+		// unlock exit lock since the go runtime doesn't reinstall
```

Comment:
> solve this instead?

OK, issue a raw system call to override the runtime and set SIG_DFL?

> Or if etcd does not call exit, it will still hang there?

This should always happen since it's closing the etcd server and etcdmain will os.Exit on etcdserver StopNotify.